pkg/database/postgres.go

```diff
@@ -0,0 +1,284 @@
+			tableName, pkConstraintName, pk))
+	}
+}
+func (v *ScriptView) writeCreateSQLForAColumn(attrType *proto.Type, tableName, attrName string,
+	primaryKeys, foreignKeyConstraints *[]string, visitedAttributes map[string]string) (string, bool) {
+	var s string
+	isAutoIncrement, isPrimaryKey := isAutoIncrementAndPrimaryKey(attrType)
+	if isPrimaryKey {
+		*primaryKeys = append(*primaryKeys, attrName)
+	}
+	if typeRef := attrType.GetTypeRef(); typeRef != nil {
+		path0 := typeRef.GetRef().Path[0]
+		path1 := typeRef.GetRef().Path[1]
+		datatype := visitedAttributes[path0+"."+path1]
+		s = fmt.Sprintf("  %s %s,\n",
+			attrName, datatype)
+		fkName := strings.ToUpper(tableName + "_" + attrName + "_FK")
+		*foreignKeyConstraints = append(
+			*foreignKeyConstraints,
+			"  CONSTRAINT "+fkName+" FOREIGN KEY("+attrName+") REFERENCES "+path0+" ("+path1+"),")
+		visitedAttributes[tableName+"."+attrName] = datatype
+	} else {
+		if isAutoIncrement {
+			s = fmt.Sprintf("  %s %s,\n", attrName, "bigserial")
+			visitedAttributes[tableName+"."+attrName] = bigIntConst
+		} else {
+			syslDataType := strings.ToLower(attrType.GetPrimitive().String())
+			var attributeSize int64
+			attributeSize = defaultTextSize
+			if syslDataType == strConst {
+				constraint := attrType.GetConstraint()
+				if len(constraint) > 0 {
+					length := constraint[0].GetLength()
+					if length != nil {
+						max := length.GetMax()
+						if max > 0 {
+							attributeSize = max
+						}
+					}
+				}
+			}
+			var datatype = v.getPostgresDataTypes(syslDataType, attributeSize)
+			s = fmt.Sprintf("  %s %s,\n", attrName, datatype)
+			visitedAttributes[tableName+"."+attrName] = datatype
+		}
+	}
+	return s, isPrimaryKey
+}
+
+func (v *ScriptView) writeModifySQLForAColumn(attrTypeOld, attrTypeNew *proto.Type, tableName,
+	attrName string, primaryKeys *[]string, visitedAttributes map[string]string) (bool, bool) {
+	typeRefNew := attrTypeNew.GetTypeRef()
+	typeRefOld := attrTypeOld.GetTypeRef()
+	primaryKeyChanged := false
+
+	isAutoIncrementOld, isPrimaryKeyOld := isAutoIncrementAndPrimaryKey(attrTypeOld)
+	isAutoIncrementNew, isPrimaryKeyNew := isAutoIncrementAndPrimaryKey(attrTypeNew)
+
+	if isPrimaryKeyNew {
+		*primaryKeys = append(*primaryKeys, attrName)
+	}
+
+	if isPrimaryKeyOld != isPrimaryKeyNew {
+		primaryKeyChanged = true
+	}
+	datatype := ""
+	fkName := strings.ToUpper(tableName + "_" + attrName + "_FK")
+	if typeRefNew != nil {
+		datatype = visitedAttributes[typeRefNew.GetRef().Path[0]+"."+typeRefNew.GetRef().Path[1]]
+		if typeRefOld == nil {
+			// typeref added. Add Foreign Key Constraint
+			v.stringBuilder.WriteString(fmt.Sprintf("ALTER TABLE %s ALTER COLUMN %s TYPE %s;\n",
+				tableName, attrName, datatype))
+			v.stringBuilder.WriteString(fmt.Sprintf(
+				"ALTER TABLE %s ADD CONSTRAINT "+fkName+" FOREIGN KEY(%s) REFERENCES %s(%s);\n",
+				tableName, attrName, typeRefNew.GetRef().Path[0], typeRefNew.GetRef().Path[1]))
+		}
+	} else {
```

Comment:
Please explore using switch cases.

Reply:
This part can not be written as switch as this is not a series of 'if' and 'else if's, but rather nested if else conditions. This function requires to process multiple scenarios and hence it has so many conditions. But those comparisons are not in a linear fashion which can be replaced with a switch.